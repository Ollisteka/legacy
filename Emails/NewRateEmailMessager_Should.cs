using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using NUnit.Framework;
using StatePrinting;

namespace Emails
{
	[TestFixture]
	public class NewRateEmailMessager_Should
	{
		private NewRateEmailMessager messager;
		private Stateprinter printer;

		[SetUp]
		public void SetUp()
		{
			messager = new NewRateEmailMessager();
			printer = new Stateprinter();
			printer.Configuration.Add(new StringBuilderConverter());
		}

		[Test]
		[UseReporter(typeof(DiffReporter))]
		public void TestCreation_When_IncreaseRate_Is_True()
		{
			Config.Local.IncreaseRate = true;
			CreateMessageAndTest();
		}

		[Test]
		[UseReporter(typeof(DiffReporter))]
		public void TestCreation_When_IncreaseRate_Is_False()
		{
			Config.Local.IncreaseRate = false;
			CreateMessageAndTest();
		}

		[Test]
		[UseReporter(typeof(DiffReporter))]
		public void TestCreation_With_DifferentRates2()
		{
			Config.Local.IncreaseRate = true;
			Config.Local.IncreaseRateFactor = 2;
			CreateMessageAndTest();
		}
		[Test]
		[UseReporter(typeof(DiffReporter))]
		public void TestCreation_With_DifferentRates1()
		{
			Config.Local.IncreaseRate = true;
			Config.Local.IncreaseRateFactor = 1;
			CreateMessageAndTest();
		}
		[Test]
		[UseReporter(typeof(DiffReporter))]
		public void TestCreation_With_DifferentRates0()
		{
			Config.Local.IncreaseRate = true;
			Config.Local.IncreaseRateFactor = 0;
			CreateMessageAndTest();
		}

		private void CreateMessageAndTest()
		{
			CombinationApprovals.VerifyAllCombinations(
				(name, type, newRate) => printer.PrintObject(messager.CreateMessage(name, type, newRate)),
				new[] { "Olga", "Evgeniy" },
				new[] { AccountType.Cheque, AccountType.Credit, AccountType.Savings },
				new[] { 0.1m, 0.01m, 0.2m });
		}
	}
}