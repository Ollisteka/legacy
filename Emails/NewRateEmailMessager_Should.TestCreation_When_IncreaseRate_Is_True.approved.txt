[Olga,Cheque,0.1] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Olga

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to 0,101%.


Kind regards - your bank.

}
[Olga,Cheque,0.01] => new EmailMessage()
{
    Important = False
    Subject = "New rate!"
    Body = Dear Olga

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to 0,0101%.


Kind regards - your bank.

}
[Olga,Cheque,0.2] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Olga

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to 0,202%.


Kind regards - your bank.

}
[Olga,Credit,0.1] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Olga

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now 0,101%


Kind regards - your bank.

}
[Olga,Credit,0.01] => new EmailMessage()
{
    Important = False
    Subject = "New rate!"
    Body = Dear Olga

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now 0,0101%


Kind regards - your bank.

}
[Olga,Credit,0.2] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Olga

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now 0,202%


Kind regards - your bank.

}
[Olga,Savings,0.1] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Olga

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to 0,101%


Kind regards - your bank.

}
[Olga,Savings,0.01] => new EmailMessage()
{
    Important = False
    Subject = "New rate!"
    Body = Dear Olga

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to 0,0101%


Kind regards - your bank.

}
[Olga,Savings,0.2] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Olga

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to 0,202%


Kind regards - your bank.

}
[Evgeniy,Cheque,0.1] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Evgeniy

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to 0,101%.


Kind regards - your bank.

}
[Evgeniy,Cheque,0.01] => new EmailMessage()
{
    Important = False
    Subject = "New rate!"
    Body = Dear Evgeniy

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to 0,0101%.


Kind regards - your bank.

}
[Evgeniy,Cheque,0.2] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Evgeniy

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to 0,202%.


Kind regards - your bank.

}
[Evgeniy,Credit,0.1] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Evgeniy

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now 0,101%


Kind regards - your bank.

}
[Evgeniy,Credit,0.01] => new EmailMessage()
{
    Important = False
    Subject = "New rate!"
    Body = Dear Evgeniy

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now 0,0101%


Kind regards - your bank.

}
[Evgeniy,Credit,0.2] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Evgeniy

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now 0,202%


Kind regards - your bank.

}
[Evgeniy,Savings,0.1] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Evgeniy

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to 0,101%


Kind regards - your bank.

}
[Evgeniy,Savings,0.01] => new EmailMessage()
{
    Important = False
    Subject = "New rate!"
    Body = Dear Evgeniy

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to 0,0101%


Kind regards - your bank.

}
[Evgeniy,Savings,0.2] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Evgeniy

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to 0,202%


Kind regards - your bank.

}
